actors/builtin/multisig/multisig_actor.go

```diff
@@ -476,33 +501,50 @@ func executeTransactionIfApproved(rt vmr.Runtime, st State, txnID TxnID, txn *Tr
 }
 
 type AddressResolveFunc func(address addr.Address) (resolved addr.Address, found bool)
+type SendFunc func(toAddr addr.Address, methodNum abi.MethodNum, params vmr.CBORMarshaler, value abi.TokenAmount) (vmr.SendReturn, exitcode.ExitCode)
 
-func isAddressEqual(resolveFunc AddressResolveFunc, addr1, addr2 addr.Address) bool {
-	return resolve(resolveFunc, addr1) == resolve(resolveFunc, addr2)
-}
-
-func isSigner(resolveFunc AddressResolveFunc, st *State, address addr.Address) bool {
-	candidateResolved := resolve(resolveFunc, address)
+func isSigner(address addr.Address, signers []addr.Address, resolveFunc AddressResolveFunc, sendFnc SendFunc) (signer bool,
+	resolved addr.Address, err error) {
+	candidateResolved, err := resolveToIDAddr(address, resolveFunc, sendFnc)
+	if err != nil {
+		return false, address, xerrors.Errorf("failed to resolve address %v to ID address: %w", address, err)
+	}
 
-	for _, ap := range st.Signers {
-		signerResolved := resolve(resolveFunc, ap)
-		if signerResolved == candidateResolved {
-			return true
+	// signer addresses have already been resolved
+	for _, ap := range signers {
+		if ap == candidateResolved {
+			return true, candidateResolved, nil
 		}
 	}
 
-	return false
+	return false, candidateResolved, nil
 }
 
-func resolve(resolveFunc AddressResolveFunc, address addr.Address) addr.Address {
-	resolved := address
-	if resolved.Protocol() != addr.ID {
-		idAddr, found := resolveFunc(resolved)
-		if found {
-			resolved = idAddr
-		}
+func resolveToIDAddr(address addr.Address, resolveFunc AddressResolveFunc, sendFunc SendFunc) (addr.Address, error) {
+	// if it's already an ID address, we can return immediately
+	if address.Protocol() == addr.ID {
+		return address, nil
 	}
```

Comment:
This short-circuit doesn't guarantee that `address` exists if it's already an ID-address.

Reply:
Plus, `resolveFunc` performs the same short-circuit internally, so I think this branch is just redundant.

Reply:
I agree. This has been removed.